### PR TITLE
Mvoebuttons

### PR DESCRIPTION
**Pull request by bitterplant**

### CODE REVIEW

File: src/test/java/cs361/battleships/models/BoardTest.java [hunk 1]
```diff
@@ -181,4 +181,18 @@ public void testGetAttack(){
...
+        board.placeShip(s, 1, 'A', false);
+        board.moveShips(1,0);
+        board.moveShips(0,1);
+        board.moveShips(-1,0);
+        board.moveShips(0,-1);
+        assertTrue(s.containsSquare(new Square(1,'A')));
+
+
+    }
+
```
Comment (freischj) [inline src/test/java/cs361/battleships/models/BoardTest.java]: Thanks for adding test cases!

File: src/main/java/cs361/battleships/models/Board.java [hunk 2]
```diff
@@ -49,20 +49,82 @@ public boolean canMove(Ship ship){
...
+                            st.moveShip(x, y);
+                            if (!canMove(st)) {
+                                st.moveShip(-x, -y);
+                            }
+                            copyIter.remove();
                         }
-                        copyIter.remove();
                     }
                 }
             }
```
Comment (freischj) [inline src/main/java/cs361/battleships/models/Board.java]: good that move right got removed out of attack

**Comment by freischj** [comment]
Perfect! seems like this finished up the project and the test cases.

**Comment by whitehug** [comment]
I haven't found any problems. Good work!

**Pull request merged**